### DISCUSSION TITLE
WPF UI x WPF collaboration

### DISCUSSION THREAD

**Discussion by pomianowski** [lepoco/wpfui#880] [Announcements]
I have the great honor to announce that in cooperation with @microsoft, especially WPF team, we will take steps to move some of the [WPF UI](https://github.com/lepoco/wpfui) functionality to the [.NET WPF](https://github.com/dotnet/wpf). 

The goal is to provide the appearance and functionality of a modern Windows 11 with the WPF technology we all know and love. Natively and effortlessly, as promised by WPF UI. It's not about moving WinUI and WindowsAppSdk to WPF, because that's what WindowsAppSdk is for. We would like to methodically expand WPF to provide a more modern look without changing habits and work style.

I am very pleased that thanks to @pchaurasia14 we will be able to contribute to WPF. I am also very pleased that @niels9001 has made efforts to make WPF UI an integral part of [PowerToys](https://github.com/microsoft/PowerToys).

This library has been developing over the last few years thanks to the support of wonderful people such as:
@IvanDmitriev1, @niels9001, @chucker, @walterlv, @ParzivalExe, @MakesYT, @Nice3point, @JeremyWu917, @timheuer, @MajorXAML  and others.

And although some of the functionality will soon be core WPF, support for older versions of .NET and some non-standard functionalities will remain the default WPF UI.

You can find the announcement from the WPF team here:
https://github.com/dotnet/wpf/discussions/8533

I'm looking forward to seeing how much interesting code we can create together.

Lots of love from Poland,
Leszek

**Comment by Nice3point** [comment]
Congratulations Leszek, you are the best

**Comment by ParzivalExe** [comment]
WOW, that is fantastic to hear. Clearly a win to everyone who works with WPF as the very inconsistent and outdates default-UI was the main reason I searched for a solution in the first place (finding this project in the process). 

Since I found it, it has been such an improvement to my WPF-Workflow that I used it as default for every project I created with WPF (even if I used an older version most of the time because this library was developing to fast for ongoing projects at points). Great to hear that at least aspects of it are now officially becoming the default for WPF.

Also, this library and the very limited work I did on it (mostly for Menus) has helped me a lot to understand more detailed parts of how to develop in WPF. I not only got a more modern version of WPF but also more knowledge about how to properly develop in WPF itself. 

Thanks for all the people who helped this project and a very big thanks to Leszek (@pomianowski) to start this project and develop it into what it is today

**Comment by EthyMoney** [comment]
Wow! This is incredible! Congratulations and thank you so much for all the effort you put into this, it makes WPF development so much nicer and it's super great to hear Microsoft is willing to step up the core WPF thanks to your efforts. This is super cool to see.

**Comment by Zenderable** [comment]
Cool! I am waiting for the results of this cooperation. 😀

**Comment by Demonese** [comment]
project reunion👍

**Comment by pjmlp** [comment]
Great news, WPF is still the best Windows GUI framework, so many features never made into UWP/WinUI.

**Reply by Qws** [reply]
And vice versa, many UWP features not made into WPF.

x:Bind, .NET Native, animation.

**Reply by chucker** [reply]
@Qws you might enjoy https://github.com/levitali/CompiledBindings

**Comment by afit** [comment]
Fantastic news — really excited for this!

**Comment by damien-c-d** [comment]
This is a big win for WPF. Great news.

**Comment by tmijail** [comment]
That's amazing!

**Comment by SkinnySackboy** [comment]
This is great.

**Comment by HClausing** [comment]
Amazing, congratulations!

**Comment by MNLierman** [comment]
Hi, I've been trying to use WPF UI, and I'm a little confused, recent comments mention that there is a version 3, and that version 2 is being maintained for minor updates? Where can I find the v3 code base and how do I get the template / demo that looks like the screenshots you've provided, because that looks nothing like the Gallery demo. Further, upon joining the Discord to get some answers about this, I found about 15+ people that have gone months without any answers, and there is bitcoin and gift card spam messages on your server. 

@pomianowski @lepoco

**Reply by niels9001** [reply]
> Hi, I've been trying to use WPF UI, and I'm a little confused, recent comments mention that there is a version 3, and that version 2 is being maintained for minor updates? Where can I find the v3 code base and how do I get the template / demo that looks like the screenshots you've provided, because that looks nothing like the Gallery demo. Further, upon joining the Discord to get some answers about this, I found about 15+ people that have gone months without any answers, and there is bitcoin and gift card spam messages on your server. 
> 
> @pomianowski @lepoco

If you want to use the 3.0 preview packages in your own app, you can check for WpfUI nuget package updates in Visual Studio (by checking the preview checkbox).

If you want to build from scratch, clone this repo, switch to the development branch and then run the Gallery project (this is the overview app you were referring to).

**Comment by shunnet** [comment]
牛逼哄哄带闪电
